lib/reqd.ml

```diff
@@ -480,14 +504,24 @@ let flush_response_body t ~max_bytes =
             ~flags:Flags.(set_end_stream default_flags)
             t.id
         in
-        (* From RFC7540§6.9.1:
-         *   Frames with zero length with the END_STREAM flag set (that is, an
-         *   empty DATA frame) MAY be sent if there is no available space in
-         *   either flow-control window. *)
-        Writer.schedule_data t.writer frame_info ~len:0 Bigstringaf.empty;
-        Writer.flush t.writer (fun () -> close_stream t);
-        stream.response_state <- Complete response;
-        0)
+        if trailers <> Headers.empty then (
+          Writer.write_response_trailers
+            t.writer
+            stream.encoder
+            frame_info
+            trailers;
+          close_stream t;
```

Comment:
the previous behavior flushed the writer before marking the stream as closed. Was this change intentional?

Reply:
Here's the recent history for this line of code
- I added the `flush` call with a `Stream.finish_stream` in https://github.com/anmonteiro/ocaml-h2/commit/d7e57959930e669bd400a80fec36ba7c72413b76#diff-71bafa60481b1abadc94a8c1599b45c6R492 (tests would fail otherwise)
- you made the `Stream.finish_stream` a `close_stream` call in https://github.com/anmonteiro/ocaml-h2/commit/d7e57959930e669bd400a80fec36ba7c72413b76#diff-71bafa60481b1abadc94a8c1599b45c6R492
- since `close_stream` already flushes (either directly, or via `reset_stream`, I removed the explicit flush call

`dune runtest` and my toy gRPC project are running fine, and a quick look at the code makes me think it should work, but I'll defer to you since you're more experienced with the codebase.

Reply:
Good point -- I forgot `close_stream` already flushed.